modules/studio-service/src/main/java/org/opencastproject/studio/endpoint/StudioEndpoint.java

```diff
@@ -0,0 +1,146 @@
+    "If the service is down or not working it will return a status 503, this means the the "
+        + "underlying service is not working and is either restarting or has failed",
+    "A status code 500 means a general failure has occurred which is not recoverable and was "
+        + "not anticipated. In other words, there is a bug! You should file an error report "
+        + "with your server logs from the time when the error occurred: "
+        + "<a href=\"https://github.com/opencast/opencast/issues\">Opencast Issue Tracker</a>" })
+@Component(immediate = true, service = StudioEndpoint.class, property = { "service.description=Studio REST Endpoint",
+    "opencast.service.type=org.opencastproject.studio", "opencast.service.path=/studio" })
+public class StudioEndpoint {
+
+  private static final Logger logger = LoggerFactory.getLogger(StudioEndpoint.class);
+
+  private final Gson gson;
+
+  private ElasticsearchIndex elasticsearchIndex;
+  private SecurityService securityService;
+
+  {
+    gson = new GsonBuilder().serializeNulls().create();
+  }
+
+  @GET
+  @Path("/series.json")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "getSeries", description = "Returns series where the current user has write permissions.",
+      returnDescription = "A list of JSON series objects",
+      restParameters = {
+         @RestParameter(name = "filter", isRequired = false,
+          description = "Usage <Filter Name>:<Value to Filter With>. Filters can combine using a comma \",\"."
+            + " Available Filters: title, textFilter.", type = STRING),
+      },
+      responses = {
+      @RestResponse(description = "Returns a list of series.",
+          responseCode = HttpServletResponse.SC_OK)
+  })
+  public Response getSeries(@QueryParam("filter") String filter) {
+    SeriesSearchQuery query = new SeriesSearchQuery(securityService.getOrganization().getId(),
+        securityService.getUser());
+    query.withoutActions();
+    query.withAction(Permissions.Action.WRITE);
+
+    if (filter != null && !filter.isBlank()) {
+      for (String f : filter.split(",")) {
+        String[] filterTuple = f.split(":",2);
+        if (filterTuple.length < 2) {
+          logger.info("Filter {} not valid: {}", filterTuple[0], filter);
+          continue;
+        }
+        String name = filterTuple[0];
+        String value = filterTuple[1];
+
+        if ("title".equals(name)) {
+          query.withTitle(value);
+        } else if ("textFilter".equals(name)) {
+          query.withText("*" + value + "*");
+        } else {
+          logger.warn("Unknown filter criteria {}", name);
+          return Response.status(SC_BAD_REQUEST).build();
+        }
+      }
+    }
+
+    try {
+      return Response.ok(gson.toJson(Arrays.stream(elasticsearchIndex.getByQuery(query).getItems())
+              .map(SearchResultItem::getSource)
+              .map(SeriesAssembler::toDto)
+              .collect(Collectors.toList())
+          )).build();
```

Comment:
Minor nit, but I would probably separate the actual searching, the gson serialization and the response building  to visually separate them from one another:

```suggestion
      var items = Arrays.stream(elasticsearchIndex.getByQuery(query).getItems())
              .map(SearchResultItem::getSource)
              .map(SeriesAssembler::toDto)
              .collect(Collectors.toList());
      var json = gson.toJson(items);
      return Response.ok(json).build();
```